Gems/Atom/Tools/MaterialEditor/gem.json

```diff
@@ -21,6 +21,7 @@
         "ImageProcessingAtom",
         "Atom_Component_DebugCamera",
         "CommonFeaturesAtom",
-        "LyShine"
+        "LyShine",
+        "O3DEMaterialEditor"
```

Comment:
This dependency of `MaterialEditor` gem with `O3DEMaterialEditor` gem can go away when the new targets `AtomMaterialEditor.Static` and `AtomMaterialEditor` are moved into their own gem.

Reply:
Instead of O3DEMaterialEditor as a new gem, my first recommendation is to use the existing view pane system without any additional layers. Just register each “material editor” as view panes and dock them in the same position.

If that is not the desirable UX, then I recommend making a more generic system and putting it in AZ tools framework. This is really a mechanism that creates a single view pane, registers that as a tool, containing a tabbed window, into which you can mount other tools. I'm not sure what you would call that system, "Tabbed Tool Docking Manager"? Have it support creating one of these tabbed tool windows by name, in this case “O3DE Material Editor”, adding tools to it, removing tools from it. Even though we are promoting events I would recommend using buses here for scriptability and consistency.

Reply:
> Instead of O3DEMaterialEditor as a new gem, my first recommendation is to use the existing view pane system without any additional layers. Just register each “material editor” as view panes and dock them in the same position.

Registering each "material editor" (render, physx, blast) separately will make the Tools menu to have 1 entry for each one. UX feedback is to reduce the number of tools, if not technically but at least the 'appearance' of having less number of tools the user have to open to do similar things.

> If that is not the desirable UX, then I recommend making a more generic system and putting it in AZ tools framework.

I like this proposal. These are my thoughts at this stage on this, the material workflow unification task is big enough and touches so many systems (atom materials, atom material editor, physx materials, blast materials) that bringing another area into the mix (editor view panel system) is going to complicate things even more, as more people will need to get involved, this new mechanism would have to be proposed, see how it fits into the bigger picture of O3DE editor tools and then get implemented.

The only difference between doing this or not is one system component and widget, which currently live inside O3DEMaterialEditor gem, but we could move them somewhere else (inside the MaterialEditor gem for example). Having this extra system component allows us to independently more forward with the material workflow unification task, while still allowing us to switch easily to a new mechanism in the future, as the only thing we'd need to do is to remove that system component.